scripts/autotools/libmongoc/CheckSSL.m4

```diff
@@ -82,10 +82,21 @@ AS_IF([test "$PHP_MONGODB_SSL" = "openssl" -o "$PHP_MONGODB_SSL" = "auto"],[
                       [have_crypto_lib="yes"],
                       [have_crypto_lib="no"],
                       [$OPENSSL_LIBDIR_LDFLAG])
+
+    have_ssl_lib="no"
+
+    dnl OpenSSL < 1.1.0
     PHP_CHECK_LIBRARY([ssl],
                       [SSL_library_init],
                       [have_ssl_lib="yes"],
-                      [have_ssl_lib="no"],
+                      [],
+                      [$OPENSSL_LIBDIR_LDFLAG -lcrypto])
+
+    dnl OpenSSL >= 1.1.0
+    PHP_CHECK_LIBRARY([ssl],
```

Comment:
Rather than always perform `PHP_CHECK_LIBRARY()` twice, I think we can improve this by wrapping the second attempt for OpenSSL >= 1.1.0 in a conditional:

```
AC_MSG_NOTICE([checking whether OpenSSL < 1.1.0 is available])
PHP_CHECK_LIBRARY([ssl],
                  [SSL_library_init],
                  [have_ssl_lib="yes"],
                  [have_ssl_lib="no"],
                  [$OPENSSL_LIBDIR_LDFLAG -lcrypto])

if test "$have_ssl_lib" = "no"; then
  AC_MSG_NOTICE([checking whether OpenSSL >= 1.1.0 is available])
  PHP_CHECK_LIBRARY([ssl],
                    [OPENSSL_init_ssl],
                    [have_ssl_lib="yes"],
                    [have_ssl_lib="no"],
                    [$OPENSSL_LIBDIR_LDFLAG -lcrypto])
fi
```

Reply:
Good call, updated

Reply:
Sorry, I missed this commit while I was working on this :-/